src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/FilterResult.cs

```diff
@@ -22,41 +22,15 @@ public FilterResult(CompletionItem completionItem, string filterText, bool match
 
         public int CompareTo(FilterResult other)
         {
-            var item1 = this.CompletionItem;
-            var item2 = other.CompletionItem;
-
-            var prefixLength1 = item1.FilterText.GetCaseInsensitivePrefixLength(this.FilterText);
-            var prefixLength2 = item2.FilterText.GetCaseInsensitivePrefixLength(other.FilterText);
-
-            // Prefer the item that matches a longer prefix of the filter text.
-            if (prefixLength1 != prefixLength2)
-            {
-                return prefixLength1.CompareTo(prefixLength2);
-            }
-
-            // If the lengths are the same, prefer the one with the higher match priority.
-            // But only if it's an item that would have been hard selected.  We don't want
-            // to aggressively select an item that was only going to be softly offered.
-            var item1Priority = item1.Rules.SelectionBehavior == CompletionItemSelectionBehavior.HardSelection
-                ? item1.Rules.MatchPriority : MatchPriority.Default;
-            var item2Priority = item2.Rules.SelectionBehavior == CompletionItemSelectionBehavior.HardSelection
-                ? item2.Rules.MatchPriority : MatchPriority.Default;
-
-            if (item1Priority != item2Priority)
-            {
-                return item1Priority.CompareTo(item2Priority);
-            }
-
-            prefixLength1 = item1.FilterText.GetCaseSensitivePrefixLength(this.FilterText);
-            prefixLength2 = item2.FilterText.GetCaseSensitivePrefixLength(other.FilterText);
-
-            // If there are "Abc" vs "abc", we should prefer the case typed by user.
-            if (prefixLength1 != prefixLength2)
-            {
-                return prefixLength1.CompareTo(prefixLength2);
-            }
-
-            return this.CompletionItem.IsPreferredItem().CompareTo(other.CompletionItem.IsPreferredItem());
+            return IComparableHelper.CompareTo(
```

Comment:
I'm a little concerned about the extra allocations here. Could this somehow end up in some hot path during completion and cause perf regression?

Reply:
yeah, agreed.  i don't see how this wouldn't be a problem.